README.rst

```diff
@@ -20,33 +20,68 @@ configurations remain synchronized with the state of all databases in the
 cluster. PgQ is used for buffering mutation records until they are collected by
 the relay and forwarded to their final destination.
 
+Function
+========
+
+``pgshovel`` is a system which replicates mutations (``INSERT``, ``UPDATE`` and ``DELETE`` operations) made to a PostgreSQL database out of the database and into an ordered stream of transaction-aware mutation messages. These messages can then be consumed by 3rd party systems as needed. Commonly, the mutation stream is consumed by data warehouse operations, cache updating services, search index services or can even be processed into other new streams.
+
+At Disqus ``pgshovel`` is actively being developed to replicate changes from our master PostgreSQLdatabase - which contains the core commenting and other relational data - into our data warehousing system. There is further planned work to use the mutation stream to power a denormalized objects database, offering highly available low-latency access to object data.
+
 Concepts
 ========
 
+``pgshovel`` has a few core concepts which are important to understanding it:
+
 Cluster
 -------
 
-A cluster contains replication sets.
+A cluster contains a collection of replication sets.
 
 Replication Set
 ---------------
 
-Replication sets contain a collection of tables that are replicated together.
+A replication set is configuration defining a collection of tables which have their mutations replicated together.
 
 Relay
 -----
 
-The relay consumes batches of mutations for the replication set and forwards
-those mutation batches to their destination.
+Relays are responsible for extracting queued raw mutations made against the tables defined in a replication set and then logging (relaying) them to a final destination. **There is exactly one relay for each replication set**.
+
+Replication
+-----------
+
+The replication system is a collection of tools which handle replicating a stream of mutation into a new target, i.e. Cassandra or S3.
+
+Within replication there are 4 main concepts:
+
+Loader
+~~~~~~
+
+A loader is responsible for dumping initial data from the tables contained in a replication set for later loading into the replication target.
 
-.. todo:: Document replication worker, loader, target.
+Stream
+~~~~~~
 
-Usage
-=====
+Stream of BatchOperation messages produced by the relay for a replication set which will eventually be replicated into the target.
```

Comment:
The part about `BatchOperation` messages is accurate now, but not sure how future proof that is if other message types are introduced. (`BatchOperation` should also be encapsulated by backticks.)